pkg/storage/etcd/etcd_helper.go

```diff
@@ -562,6 +613,9 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 		if err != nil {
 			return err
 		}
+		if err := checkPreconditionsMatch(preconditions, obj); err != nil {
```

Comment:
Hm, I'm not sure about this.

1. I was expecting this to be enforced in the tryUpdate function?
2. I was expecting an error here to cause a retry (up to some number).

But I'm still thinking, maybe this is fine.